// @activepieces/shared/src/lib/issues/issue-dto.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PopulatedIssue = exports.Issue = exports.IssueStatus = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_1 = require("../common");
const id_generator_1 = require("../common/id-generator");
var IssueStatus;
(function (IssueStatus) {
    IssueStatus["RESOLVED"] = "RESOLVED";
    IssueStatus["UNRESOLVED"] = "UNRESOLVED";
    IssueStatus["ARCHIVED"] = "ARCHIVED";
})(IssueStatus || (exports.IssueStatus = IssueStatus = {}));
const Step = typebox_1.Type.Object({
    stepName: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
});
exports.Issue = typebox_1.Type.Object(Object.assign(Object.assign({}, common_1.BaseModelSchema), { projectId: id_generator_1.ApId, flowId: id_generator_1.ApId, status: typebox_1.Type.Enum(IssueStatus), stepName: typebox_1.Type.Optional(typebox_1.Type.String()), lastOccurrence: typebox_1.Type.String() }));
exports.PopulatedIssue = typebox_1.Type.Composite([exports.Issue, typebox_1.Type.Object({
        count: typebox_1.Type.Number({ default: 0 }),
        step: typebox_1.Type.Optional(Step),
        flowDisplayName: typebox_1.Type.String(),
    })]);
