// @activepieces/shared/src/lib/common/security/permission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleType = exports.Permission = void 0;
var Permission;
(function (Permission) {
    Permission["READ_APP_CONNECTION"] = "READ_APP_CONNECTION";
    Permission["WRITE_APP_CONNECTION"] = "WRITE_APP_CONNECTION";
    Permission["READ_FLOW"] = "READ_FLOW";
    Permission["WRITE_FLOW"] = "WRITE_FLOW";
    Permission["UPDATE_FLOW_STATUS"] = "UPDATE_FLOW_STATUS";
    Permission["WRITE_INVITATION"] = "WRITE_INVITATION";
    Permission["READ_INVITATION"] = "READ_INVITATION";
    Permission["READ_PROJECT_MEMBER"] = "READ_PROJECT_MEMBER";
    Permission["WRITE_PROJECT_MEMBER"] = "WRITE_PROJECT_MEMBER";
    Permission["WRITE_GIT_REPO"] = "WRITE_GIT_REPO";
    Permission["READ_GIT_REPO"] = "READ_GIT_REPO";
    Permission["READ_RUN"] = "READ_RUN";
    Permission["WRITE_RUN"] = "WRITE_RUN";
    Permission["READ_ISSUES"] = "READ_ISSUES";
    Permission["WRITE_ISSUES"] = "WRITE_ISSUES";
    Permission["READ_FOLDER"] = "READ_FOLDER";
    Permission["WRITE_FOLDER"] = "WRITE_FOLDER";
    Permission["WRITE_ALERT"] = "WRITE_ALERT";
    Permission["READ_ALERT"] = "READ_ALERT";
    Permission["WRITE_PROJECT"] = "WRITE_PROJECT";
    Permission["READ_PROJECT"] = "READ_PROJECT";
})(Permission || (exports.Permission = Permission = {}));
var RoleType;
(function (RoleType) {
    RoleType["DEFAULT"] = "DEFAULT";
    RoleType["CUSTOM"] = "CUSTOM";
})(RoleType || (exports.RoleType = RoleType = {}));
