// @activepieces/pieces-common/src/lib/ai/providers/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hasMapper = exports.ModelType = void 0;
exports.authHeader = authHeader;
exports.model = model;
exports.imageMapper = imageMapper;
exports.chatMapper = chatMapper;
function authHeader(options) {
    var _a;
    return {
        name: options.bearer ? 'Authorization' : options.name,
        mapper: options.bearer
            ? headerValueMappers.bearer
            : (_a = options.mapper) !== null && _a !== void 0 ? _a : headerValueMappers.default,
    };
}
const headerValueMappers = {
    bearer: (value) => `Bearer ${value}`,
    default: (value) => value,
};
var ModelType;
(function (ModelType) {
    ModelType["IMAGE"] = "image";
    ModelType["CHAT"] = "chat";
    ModelType["NO_MAPPER"] = "no-mapper";
})(ModelType || (exports.ModelType = ModelType = {}));
function model(options) {
    return {
        __tag: ModelType.NO_MAPPER,
        label: options.label,
        value: options.value,
        supported: options.supported,
        mapper: function (mapper) {
            return Object.assign(Object.assign({}, this), { mapper: mapper, __tag: mapper.__tag });
        },
    };
}
function imageMapper(mapper) {
    return Object.assign(Object.assign({}, mapper), { __tag: ModelType.IMAGE });
}
function chatMapper(mapper) {
    return Object.assign(Object.assign({}, mapper), { __tag: ModelType.CHAT });
}
const hasMapper = (m) => {
    return m.__tag === ModelType.IMAGE || m.__tag === ModelType.CHAT;
};
exports.hasMapper = hasMapper;
