// @activepieces/shared/src/lib/mcp/tools/mcp-tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpTool = exports.McpPieceToolData = exports.McpToolType = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_1 = require("../../common");
const id_generator_1 = require("../../common/id-generator");
const flows_1 = require("../../flows");
var McpToolType;
(function (McpToolType) {
    McpToolType["PIECE"] = "PIECE";
    McpToolType["FLOW"] = "FLOW";
})(McpToolType || (exports.McpToolType = McpToolType = {}));
exports.McpPieceToolData = typebox_1.Type.Object({
    pieceName: typebox_1.Type.String(),
    pieceVersion: typebox_1.Type.String(),
    actionNames: typebox_1.Type.Array(typebox_1.Type.String()),
    logoUrl: typebox_1.Type.String(),
    connectionExternalId: typebox_1.Type.Optional(typebox_1.Type.String()),
});
exports.McpTool = typebox_1.Type.Object(Object.assign(Object.assign({}, common_1.BaseModelSchema), { type: typebox_1.Type.Enum(McpToolType), mcpId: id_generator_1.ApId, pieceMetadata: typebox_1.Type.Optional(exports.McpPieceToolData), flowId: typebox_1.Type.Optional(id_generator_1.ApId), flow: typebox_1.Type.Optional(flows_1.PopulatedFlow) }));
