// @activepieces/piece-github/src/lib/actions/create-comment-on-a-issue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubCreateCommentOnAIssue = void 0;
const tslib_1 = require("tslib");
const index_1 = require("../../index");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const common_1 = require("../common");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubCreateCommentOnAIssue = (0, pieces_framework_1.createAction)({
    auth: index_1.githubAuth,
    name: 'createCommentOnAIssue',
    displayName: 'Create comment on a issue',
    description: 'Adds a comment to the specified issue (also works with pull requests)',
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
        issue_number: pieces_framework_1.Property.Number({
            displayName: 'Issue number',
            description: 'The number of the issue to comment on',
            required: true,
        }),
        comment: pieces_framework_1.Property.LongText({
            displayName: 'Comment',
            description: 'The comment to add to the issue',
            required: true,
        }),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const issue_number = propsValue.issue_number;
            const { owner, repo } = propsValue.repository;
            const response = yield (0, common_1.githubApiCall)({
                accessToken: auth.access_token,
                method: pieces_common_1.HttpMethod.POST,
                resourceUri: `/repos/${owner}/${repo}/issues/${issue_number}/comments`,
                body: {
                    body: propsValue.comment,
                },
            });
            return response;
        });
    },
});
