// @activepieces/piece-github/src/lib/actions/create-issue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubCreateIssueAction = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const __1 = require("../../");
const common_1 = require("../common");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubCreateIssueAction = (0, pieces_framework_1.createAction)({
    auth: __1.githubAuth,
    name: 'github_create_issue',
    displayName: 'Create Issue',
    description: 'Create Issue in GitHub Repository',
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
        title: pieces_framework_1.Property.ShortText({
            displayName: 'Title',
            description: 'The title of the issue',
            required: true,
        }),
        description: pieces_framework_1.Property.LongText({
            displayName: 'Description',
            description: 'The description of the issue',
            required: false,
        }),
        labels: common_1.githubCommon.labelDropDown(),
        assignees: common_1.githubCommon.assigneeDropDown(),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const { title, assignees, labels, description } = propsValue;
            const { owner, repo } = propsValue.repository;
            const issueFields = {
                title,
                body: description,
            };
            if (labels) {
                issueFields['labels'] = labels;
            }
            if (assignees) {
                issueFields['assignees'] = assignees;
            }
            const response = yield (0, common_1.githubApiCall)({
                accessToken: auth.access_token,
                method: pieces_common_1.HttpMethod.POST,
                resourceUri: `/repos/${owner}/${repo}/issues`,
                body: issueFields,
            });
            return response;
        });
    },
});
