// @activepieces/piece-github/src/lib/actions/get-issue-information.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubGetIssueInformation = void 0;
const tslib_1 = require("tslib");
const index_1 = require("../../index");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const common_1 = require("../common");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubGetIssueInformation = (0, pieces_framework_1.createAction)({
    auth: index_1.githubAuth,
    name: 'getIssueInformation',
    displayName: 'Get issue information',
    description: 'Grabs information from a specific issue',
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
        issue_number: pieces_framework_1.Property.Number({
            displayName: 'Issue Number',
            description: 'The number of the issue you want to get information from',
            required: true,
        }),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const issue_number = propsValue.issue_number;
            const { owner, repo } = propsValue.repository;
            const response = yield (0, common_1.githubApiCall)({
                accessToken: auth.access_token,
                method: pieces_common_1.HttpMethod.GET,
                resourceUri: `/repos/${owner}/${repo}/issues/${issue_number}`,
            });
            return response;
        });
    },
});
