// @activepieces/piece-github/src/lib/actions/lock-issue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubLockIssueAction = void 0;
const tslib_1 = require("tslib");
const index_1 = require("../../index");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const common_1 = require("../common");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubLockIssueAction = (0, pieces_framework_1.createAction)({
    auth: index_1.githubAuth,
    name: 'lockIssue',
    displayName: 'Lock issue',
    description: 'Locks the specified issue',
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
        issue_number: pieces_framework_1.Property.Number({
            displayName: 'Issue Number',
            description: 'The number of the issue to be locked',
            required: true,
        }),
        lock_reason: pieces_framework_1.Property.Dropdown({
            displayName: 'Lock Reason',
            description: 'The reason for locking the issue',
            required: false,
            refreshers: [],
            options: () => tslib_1.__awaiter(void 0, void 0, void 0, function* () {
                return {
                    options: [
                        { value: 'off-topic', label: 'Off-topic' },
                        { value: 'too heated', label: 'Too heated' },
                        { value: 'resolved', label: 'Resolved' },
                        { value: 'spam', label: 'Spam' },
                    ],
                };
            }),
        }),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const { issue_number, lock_reason } = propsValue;
            const { owner, repo } = propsValue.repository;
            const response = yield (0, common_1.githubApiCall)({
                accessToken: auth.access_token,
                method: pieces_common_1.HttpMethod.PUT,
                resourceUri: `/repos/${owner}/${repo}/issues/${issue_number}/lock`,
                body: {
                    lock_reason,
                },
            });
            return response;
        });
    },
});
