// @activepieces/piece-github/src/lib/actions/raw-graphql-query.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubRawGraphqlQuery = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const __1 = require("../..");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubRawGraphqlQuery = (0, pieces_framework_1.createAction)({
    name: 'rawGraphqlQuery',
    displayName: 'Raw GraphQL query',
    description: 'Perform a raw GraphQL query',
    auth: __1.githubAuth,
    props: {
        query: pieces_framework_1.Property.LongText({ displayName: 'Query', required: true }),
        variables: pieces_framework_1.Property.Object({ displayName: 'Parameters', required: false }),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const response = yield pieces_common_1.httpClient.sendRequest({
                url: 'https://api.github.com/graphql',
                method: pieces_common_1.HttpMethod.POST,
                body: JSON.stringify({
                    query: propsValue.query,
                    variables: propsValue.variables,
                }),
                authentication: {
                    type: pieces_common_1.AuthenticationType.BEARER_TOKEN,
                    token: auth.access_token,
                },
            });
            return response;
        });
    },
});
