// @activepieces/piece-github/src/lib/actions/unlock-issue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubUnlockIssueAction = void 0;
const tslib_1 = require("tslib");
const index_1 = require("../../index");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const common_1 = require("../common");
const pieces_common_1 = require("@activepieces/pieces-common");
exports.githubUnlockIssueAction = (0, pieces_framework_1.createAction)({
    auth: index_1.githubAuth,
    name: 'unlockIssue',
    displayName: 'Unlock issue',
    description: 'Unlocks the specified issue',
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
        issue_number: pieces_framework_1.Property.Number({
            displayName: 'Issue Number',
            description: 'The number of the issue to be unlocked',
            required: true,
        }),
    },
    run(_a) {
        return tslib_1.__awaiter(this, arguments, void 0, function* ({ auth, propsValue }) {
            const { issue_number } = propsValue;
            const { owner, repo } = propsValue.repository;
            const response = yield (0, common_1.githubApiCall)({
                accessToken: auth.access_token,
                method: pieces_common_1.HttpMethod.DELETE,
                resourceUri: `/repos/${owner}/${repo}/issues/${issue_number}/lock`,
            });
            return response;
        });
    },
});
