// @activepieces/shared/src/lib/flows/flow.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PopulatedFlow = exports.Flow = exports.FlowScheduleOptions = exports.WebhookHandshakeConfiguration = exports.WebhookHandshakeStrategy = exports.FlowStatus = exports.ScheduleType = void 0;
const typebox_1 = require("@sinclair/typebox");
const base_model_1 = require("../common/base-model");
const metadata_1 = require("../common/metadata");
const flow_version_1 = require("./flow-version");
var ScheduleType;
(function (ScheduleType) {
    ScheduleType["CRON_EXPRESSION"] = "CRON_EXPRESSION";
})(ScheduleType || (exports.ScheduleType = ScheduleType = {}));
var FlowStatus;
(function (FlowStatus) {
    FlowStatus["ENABLED"] = "ENABLED";
    FlowStatus["DISABLED"] = "DISABLED";
})(FlowStatus || (exports.FlowStatus = FlowStatus = {}));
var WebhookHandshakeStrategy;
(function (WebhookHandshakeStrategy) {
    WebhookHandshakeStrategy["NONE"] = "NONE";
    WebhookHandshakeStrategy["HEADER_PRESENT"] = "HEADER_PRESENT";
    WebhookHandshakeStrategy["QUERY_PRESENT"] = "QUERY_PRESENT";
    WebhookHandshakeStrategy["BODY_PARAM_PRESENT"] = "BODY_PARAM_PRESENT";
})(WebhookHandshakeStrategy || (exports.WebhookHandshakeStrategy = WebhookHandshakeStrategy = {}));
exports.WebhookHandshakeConfiguration = typebox_1.Type.Object({
    strategy: typebox_1.Type.Enum(WebhookHandshakeStrategy),
    paramName: typebox_1.Type.Optional(typebox_1.Type.String()),
});
exports.FlowScheduleOptions = typebox_1.Type.Object({
    type: typebox_1.Type.Literal(ScheduleType.CRON_EXPRESSION),
    cronExpression: typebox_1.Type.String(),
    timezone: typebox_1.Type.String(),
    failureCount: typebox_1.Type.Optional(typebox_1.Type.Number()),
});
exports.Flow = typebox_1.Type.Object(Object.assign(Object.assign({}, base_model_1.BaseModelSchema), { projectId: typebox_1.Type.String(), externalId: typebox_1.Type.String(), folderId: (0, base_model_1.Nullable)(typebox_1.Type.String()), status: typebox_1.Type.Enum(FlowStatus), schedule: (0, base_model_1.Nullable)(exports.FlowScheduleOptions), handshakeConfiguration: (0, base_model_1.Nullable)(exports.WebhookHandshakeConfiguration), publishedVersionId: (0, base_model_1.Nullable)(typebox_1.Type.String()), metadata: (0, base_model_1.Nullable)(metadata_1.Metadata) }));
exports.PopulatedFlow = typebox_1.Type.Composite([
    exports.Flow,
    typebox_1.Type.Object({
        version: flow_version_1.FlowVersion,
    }),
]);
