// @activepieces/shared/src/lib/todos/content.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RichContentBlock = exports.ToolCallContentBlock = exports.ToolCallType = exports.ToolCallStatus = exports.MarkdownContentBlock = exports.ContentBlockType = void 0;
const typebox_1 = require("@sinclair/typebox");
const base_model_1 = require("../common/base-model");
var ContentBlockType;
(function (ContentBlockType) {
    ContentBlockType["MARKDOWN"] = "markdown";
    ContentBlockType["TOOL_CALL"] = "tool-call";
})(ContentBlockType || (exports.ContentBlockType = ContentBlockType = {}));
exports.MarkdownContentBlock = typebox_1.Type.Object({
    type: typebox_1.Type.Literal(ContentBlockType.MARKDOWN),
    markdown: typebox_1.Type.String(),
});
var ToolCallStatus;
(function (ToolCallStatus) {
    ToolCallStatus["IN_PROGRESS"] = "in-progress";
    ToolCallStatus["COMPLETED"] = "completed";
})(ToolCallStatus || (exports.ToolCallStatus = ToolCallStatus = {}));
var ToolCallType;
(function (ToolCallType) {
    ToolCallType["PIECE"] = "piece";
    ToolCallType["FLOW"] = "flow";
})(ToolCallType || (exports.ToolCallType = ToolCallType = {}));
exports.ToolCallContentBlock = typebox_1.Type.Object({
    type: typebox_1.Type.Literal(ContentBlockType.TOOL_CALL),
    displayName: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
    toolCallType: typebox_1.Type.Enum(ToolCallType),
    logoUrl: (0, base_model_1.Nullable)(typebox_1.Type.String()),
    status: typebox_1.Type.Enum(ToolCallStatus),
    input: (0, base_model_1.Nullable)(typebox_1.Type.Unknown()),
    output: (0, base_model_1.Nullable)(typebox_1.Type.Unknown()),
    startTime: typebox_1.Type.String(),
    endTime: typebox_1.Type.Optional(typebox_1.Type.String()),
});
exports.RichContentBlock = typebox_1.Type.Union([exports.MarkdownContentBlock, exports.ToolCallContentBlock]);
