// @activepieces/shared/src/lib/websocket/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebsocketServerEvent = exports.TodoActivityCreated = exports.TodoActivityChanged = exports.TodoChanged = exports.WebsocketClientEvent = void 0;
const typebox_1 = require("@sinclair/typebox");
const content_1 = require("../todos/content");
var WebsocketClientEvent;
(function (WebsocketClientEvent) {
    WebsocketClientEvent["TEST_FLOW_RUN_STARTED"] = "TEST_FLOW_RUN_STARTED";
    WebsocketClientEvent["ASK_COPILOT_FINISHED"] = "ASK_COPILOT_FINISHED";
    WebsocketClientEvent["TEST_STEP_FINISHED"] = "TEST_STEP_FINISHED";
    WebsocketClientEvent["REFRESH_PIECE"] = "REFRESH_PIECE";
    WebsocketClientEvent["FLOW_RUN_PROGRESS"] = "FLOW_RUN_PROGRESS";
    WebsocketClientEvent["TODO_CHANGED"] = "TODO_CHANGED";
    WebsocketClientEvent["TODO_ACTIVITY_CHANGED"] = "TODO_ACTIVITY_CHANGED";
    WebsocketClientEvent["TODO_ACTIVITY_CREATED"] = "TODO_ACTIVITY_CREATED";
})(WebsocketClientEvent || (exports.WebsocketClientEvent = WebsocketClientEvent = {}));
exports.TodoChanged = typebox_1.Type.Object({
    todoId: typebox_1.Type.String(),
});
exports.TodoActivityChanged = typebox_1.Type.Object({
    activityId: typebox_1.Type.String(),
    todoId: typebox_1.Type.String(),
    content: typebox_1.Type.Array(content_1.RichContentBlock),
});
exports.TodoActivityCreated = typebox_1.Type.Object({
    todoId: typebox_1.Type.String(),
});
var WebsocketServerEvent;
(function (WebsocketServerEvent) {
    WebsocketServerEvent["TEST_STEP_RUN"] = "TEST_STEP_RUN";
    WebsocketServerEvent["ASK_COPILOT"] = "ASK_COPILOT";
    WebsocketServerEvent["TEST_FLOW_RUN"] = "TEST_FLOW_RUN";
})(WebsocketServerEvent || (exports.WebsocketServerEvent = WebsocketServerEvent = {}));
