// @activepieces/shared/src/lib/ai/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIUsage = exports.CreateAIProviderRequest = exports.AIProviderWithoutSensitiveData = exports.AIProvider = void 0;
const tslib_1 = require("tslib");
const typebox_1 = require("@sinclair/typebox");
const common_1 = require("../common");
exports.AIProvider = typebox_1.Type.Object(Object.assign(Object.assign({}, common_1.BaseModelSchema), { config: typebox_1.Type.Object({
        apiKey: typebox_1.Type.String(),
    }), provider: typebox_1.Type.String({ minLength: 1 }), platformId: typebox_1.Type.String() }));
exports.AIProviderWithoutSensitiveData = typebox_1.Type.Omit(exports.AIProvider, ['config']);
exports.CreateAIProviderRequest = typebox_1.Type.Object({
    provider: typebox_1.Type.String({ minLength: 1 }),
    apiKey: typebox_1.Type.String(),
});
exports.AIUsage = typebox_1.Type.Object(Object.assign(Object.assign({}, common_1.BaseModelSchema), { provider: typebox_1.Type.String({ minLength: 1 }), model: typebox_1.Type.String({ minLength: 1 }), cost: typebox_1.Type.Number({ minimum: 0 }), projectId: typebox_1.Type.String() }));
tslib_1.__exportStar(require("./supported-ai-providers"), exports);
tslib_1.__exportStar(require("./ai-sdk"), exports);
