// @activepieces/piece-notion/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.notion = exports.notionAuth = void 0;
const tslib_1 = require("tslib");
const pieces_common_1 = require("@activepieces/pieces-common");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const shared_1 = require("@activepieces/shared");
const append_to_page_1 = require("./lib/actions/append-to-page");
const create_database_item_1 = require("./lib/actions/create-database-item");
const create_page_1 = require("./lib/actions/create-page");
const update_database_item_1 = require("./lib/actions/update-database-item");
const new_database_item_1 = require("./lib/triggers/new-database-item");
const updated_database_item_1 = require("./lib/triggers/updated-database-item");
const find_item_1 = require("./lib/actions/find-item");
const get_page_or_block_children_1 = require("./lib/actions/get-page-or-block-children");
exports.notionAuth = pieces_framework_1.PieceAuth.OAuth2({
    authUrl: 'https://api.notion.com/v1/oauth/authorize',
    tokenUrl: 'https://api.notion.com/v1/oauth/token',
    scope: [],
    extra: {
        owner: 'user',
    },
    authorizationMethod: pieces_framework_1.OAuth2AuthorizationMethod.HEADER,
    required: true,
});
exports.notion = (0, pieces_framework_1.createPiece)({
    displayName: 'Notion',
    description: 'The all-in-one workspace',
    logoUrl: 'https://cdn.activepieces.com/pieces/notion.png',
    categories: [shared_1.PieceCategory.PRODUCTIVITY],
    minimumSupportedRelease: '0.30.0',
    authors: [
        'ShayPunter',
        'kishanprmr',
        'MoShizzle',
        'khaledmashaly',
        'abuaboud',
        'AdamSelene',
    ],
    auth: exports.notionAuth,
    actions: [
        create_database_item_1.createDatabaseItem,
        update_database_item_1.updateDatabaseItem,
        find_item_1.findDatabaseItem,
        create_page_1.createPage,
        append_to_page_1.appendToPage,
        get_page_or_block_children_1.getPageOrBlockChildren,
        (0, pieces_common_1.createCustomApiCallAction)({
            baseUrl: () => 'https://api.notion.com/v1',
            auth: exports.notionAuth,
            authMapping: (auth) => tslib_1.__awaiter(void 0, void 0, void 0, function* () {
                return ({
                    Authorization: `Bearer ${auth.access_token}`,
                });
            }),
        }),
    ],
    triggers: [new_database_item_1.newDatabaseItem, updated_database_item_1.updatedDatabaseItem],
});
