// @activepieces/piece-github/src/lib/trigger/register-trigger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.githubRegisterTrigger = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const common_1 = require("../common");
const __1 = require("../../");
const pieces_common_1 = require("@activepieces/pieces-common");
const githubRegisterTrigger = ({ name, displayName, description, sampleData, }) => (0, pieces_framework_1.createTrigger)({
    auth: __1.githubAuth,
    name: `trigger_${name}`,
    displayName,
    description,
    props: {
        repository: common_1.githubCommon.repositoryDropdown,
    },
    sampleData,
    type: pieces_framework_1.TriggerStrategy.WEBHOOK,
    onEnable(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const { repo, owner } = context.propsValue.repository;
            const response = yield (0, common_1.githubApiCall)({
                accessToken: context.auth.access_token,
                method: pieces_common_1.HttpMethod.POST,
                resourceUri: `/repos/${owner}/${repo}/hooks`,
                body: {
                    name: 'web',
                    active: true,
                    events: [name],
                    config: {
                        url: context.webhookUrl,
                        content_type: 'json',
                        insecure_ssl: '0',
                    },
                },
            });
            yield context.store.put(`github_${name}_trigger`, {
                webhookId: response.body.id,
                owner: owner,
                repo: repo,
            });
        });
    },
    onDisable(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const response = yield context.store.get(`github_${name}_trigger`);
            if (response !== null && response !== undefined) {
                yield (0, common_1.githubApiCall)({
                    accessToken: context.auth.access_token,
                    method: pieces_common_1.HttpMethod.DELETE,
                    resourceUri: `/repos/${response.owner}/${response.repo}/hooks/${response.webhookId}`,
                });
            }
        });
    },
    run(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            console.debug('payload received', context.payload.body);
            if (isVerificationCall(context.payload.body)) {
                return [];
            }
            return [context.payload.body];
        });
    },
});
exports.githubRegisterTrigger = githubRegisterTrigger;
function isVerificationCall(payload) {
    return payload['zen'] !== undefined;
}
