// @activepieces/piece-notion/src/lib/actions/append-to-page.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.appendToPage = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const client_1 = require("@notionhq/client");
const __1 = require("../..");
const common_1 = require("../common");
const martian_1 = require("@tryfabric/martian");
exports.appendToPage = (0, pieces_framework_1.createAction)({
    auth: __1.notionAuth,
    name: 'append_to_page',
    displayName: 'Append to Page',
    description: 'Appends content to the end of a page.',
    props: {
        pageId: common_1.notionCommon.page,
        content: pieces_framework_1.Property.LongText({
            displayName: 'Content',
            description: 'The content you want to append. You can use markdown formatting.',
            required: true,
        }),
    },
    run(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const { pageId, content } = context.propsValue;
            const notion = new client_1.Client({
                auth: context.auth.access_token,
                notionVersion: '2022-02-22',
            });
            return yield notion.blocks.children.append({
                block_id: pageId,
                children: (0, martian_1.markdownToBlocks)(content),
            });
        });
    },
});
