// @activepieces/piece-notion/src/lib/actions/create-page.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPage = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const client_1 = require("@notionhq/client");
const __1 = require("../..");
const common_1 = require("../common");
exports.createPage = (0, pieces_framework_1.createAction)({
    auth: __1.notionAuth,
    name: 'createPage',
    displayName: 'Create Page',
    description: 'Create a page under a parent page.',
    props: {
        pageId: common_1.notionCommon.page,
        title: pieces_framework_1.Property.ShortText({
            displayName: 'Title',
            description: 'The title of the page.',
            required: false,
        }),
        content: pieces_framework_1.Property.LongText({
            displayName: 'Content',
            description: 'The content of the page.',
            required: false,
        }),
    },
    run(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const { pageId, title, content } = context.propsValue;
            const notion = new client_1.Client({
                auth: context.auth.access_token,
                notionVersion: '2022-02-22',
            });
            const pageProperties = {
                title: {
                    title: [
                        {
                            text: {
                                content: title !== null && title !== void 0 ? title : '',
                            },
                        },
                    ],
                },
            };
            const children = [];
            // Add content to page
            if (content)
                children.push({
                    object: 'block',
                    type: 'paragraph',
                    paragraph: {
                        rich_text: [
                            {
                                type: 'text',
                                text: {
                                    content: content,
                                },
                            },
                        ],
                    },
                });
            const page = yield notion.pages.create({
                parent: {
                    page_id: pageId,
                },
                properties: pageProperties,
                children: children,
            });
            return page;
        });
    },
});
