// @activepieces/piece-notion/src/lib/actions/update-database-item.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateDatabaseItem = void 0;
const tslib_1 = require("tslib");
const pieces_framework_1 = require("@activepieces/pieces-framework");
const client_1 = require("@notionhq/client");
const models_1 = require("../common/models");
const __1 = require("../..");
const common_1 = require("../common");
exports.updateDatabaseItem = (0, pieces_framework_1.createAction)({
    auth: __1.notionAuth,
    name: 'update_database_item',
    displayName: 'Update Database Item',
    description: 'Updates an item in database',
    props: {
        database_id: common_1.notionCommon.database_id,
        database_item_id: common_1.notionCommon.database_item_id,
        databaseFields: common_1.notionCommon.databaseFields,
    },
    run(context) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const { database_id, database_item_id, databaseFields } = context.propsValue;
            if (!database_item_id)
                throw new Error('Item ID is required');
            const notionFields = {};
            const notion = new client_1.Client({
                auth: context.auth.access_token,
                notionVersion: '2022-02-22',
            });
            const { properties } = yield notion.databases.retrieve({
                database_id: database_id,
            });
            Object.keys(databaseFields).forEach((key) => {
                if (databaseFields[key] !== '') {
                    const fieldType = properties[key].type;
                    notionFields[key] = models_1.NotionFieldMapping[fieldType].buildNotionType(databaseFields[key]);
                }
            });
            return yield notion.pages.update({
                page_id: database_item_id,
                properties: notionFields,
            });
        });
    },
});
