// @activepieces/shared/src/lib/project-release/project-state.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectSyncPlan = exports.ProjectSyncPlanOperation = exports.ProjectSyncError = exports.DiffState = exports.TableOperation = exports.ConnectionOperation = exports.ProjectOperation = exports.ProjectState = exports.TableState = exports.FieldState = exports.ConnectionState = exports.TableOperationType = exports.FlowState = exports.ConnectionOperationType = exports.ProjectOperationType = void 0;
const typebox_1 = require("@sinclair/typebox");
const flow_1 = require("../flows/flow");
var ProjectOperationType;
(function (ProjectOperationType) {
    ProjectOperationType["UPDATE_FLOW"] = "UPDATE_FLOW";
    ProjectOperationType["CREATE_FLOW"] = "CREATE_FLOW";
    ProjectOperationType["DELETE_FLOW"] = "DELETE_FLOW";
})(ProjectOperationType || (exports.ProjectOperationType = ProjectOperationType = {}));
var ConnectionOperationType;
(function (ConnectionOperationType) {
    ConnectionOperationType["UPDATE_CONNECTION"] = "UPDATE_CONNECTION";
    ConnectionOperationType["CREATE_CONNECTION"] = "CREATE_CONNECTION";
})(ConnectionOperationType || (exports.ConnectionOperationType = ConnectionOperationType = {}));
exports.FlowState = flow_1.PopulatedFlow;
var TableOperationType;
(function (TableOperationType) {
    TableOperationType["UPDATE_TABLE"] = "UPDATE_TABLE";
    TableOperationType["CREATE_TABLE"] = "CREATE_TABLE";
})(TableOperationType || (exports.TableOperationType = TableOperationType = {}));
exports.ConnectionState = typebox_1.Type.Object({
    externalId: typebox_1.Type.String(),
    pieceName: typebox_1.Type.String(),
    displayName: typebox_1.Type.String(),
});
exports.FieldState = typebox_1.Type.Object({
    name: typebox_1.Type.String(),
    type: typebox_1.Type.String(),
    data: typebox_1.Type.Optional(typebox_1.Type.Object({
        options: typebox_1.Type.Array(typebox_1.Type.Object({
            value: typebox_1.Type.String(),
        })),
    })),
    externalId: typebox_1.Type.String(),
});
exports.TableState = typebox_1.Type.Object({
    id: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
    externalId: typebox_1.Type.String(),
    fields: typebox_1.Type.Array(exports.FieldState),
});
exports.ProjectState = typebox_1.Type.Object({
    flows: typebox_1.Type.Array(flow_1.PopulatedFlow),
    // NOTE: This is optional because in old releases, the connections and tables state is not present
    connections: typebox_1.Type.Optional(typebox_1.Type.Array(exports.ConnectionState)),
    tables: typebox_1.Type.Optional(typebox_1.Type.Array(exports.TableState)),
});
exports.ProjectOperation = typebox_1.Type.Union([
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.UPDATE_FLOW),
        newFlowState: exports.FlowState,
        flowState: exports.FlowState,
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.CREATE_FLOW),
        flowState: exports.FlowState,
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.DELETE_FLOW),
        flowState: exports.FlowState,
    }),
]);
exports.ConnectionOperation = typebox_1.Type.Union([
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ConnectionOperationType.UPDATE_CONNECTION),
        newConnectionState: exports.ConnectionState,
        connectionState: exports.ConnectionState,
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ConnectionOperationType.CREATE_CONNECTION),
        connectionState: exports.ConnectionState,
    }),
]);
exports.TableOperation = typebox_1.Type.Union([
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(TableOperationType.UPDATE_TABLE),
        newTableState: exports.TableState,
        tableState: exports.TableState,
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(TableOperationType.CREATE_TABLE),
        tableState: exports.TableState,
    }),
]);
exports.DiffState = typebox_1.Type.Object({
    operations: typebox_1.Type.Array(exports.ProjectOperation),
    connections: typebox_1.Type.Array(exports.ConnectionOperation),
    tables: typebox_1.Type.Array(exports.TableOperation),
});
exports.ProjectSyncError = typebox_1.Type.Object({
    flowId: typebox_1.Type.String(),
    message: typebox_1.Type.String(),
});
exports.ProjectSyncPlanOperation = typebox_1.Type.Union([
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.CREATE_FLOW),
        flow: typebox_1.Type.Object({
            id: typebox_1.Type.String(),
            displayName: typebox_1.Type.String(),
        }),
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.UPDATE_FLOW),
        flow: typebox_1.Type.Object({
            id: typebox_1.Type.String(),
            displayName: typebox_1.Type.String(),
        }),
        targetFlow: typebox_1.Type.Object({
            id: typebox_1.Type.String(),
            displayName: typebox_1.Type.String(),
        }),
    }),
    typebox_1.Type.Object({
        type: typebox_1.Type.Literal(ProjectOperationType.DELETE_FLOW),
        flow: typebox_1.Type.Object({
            id: typebox_1.Type.String(),
            displayName: typebox_1.Type.String(),
        }),
    }),
]);
exports.ProjectSyncPlan = typebox_1.Type.Object({
    operations: typebox_1.Type.Array(exports.ProjectSyncPlanOperation),
    connections: typebox_1.Type.Array(exports.ConnectionOperation),
    tables: typebox_1.Type.Array(exports.TableOperation),
    errors: typebox_1.Type.Array(exports.ProjectSyncError),
});
